RFC/src/RFC-0201_TariScript.md

```diff
@@ -0,0 +1,410 @@
+We assume that the rules were properly enforced up until now, but it's not really our concern. What we care about is
+that no inflation has occurred to this point and that we enforce all the consensus rules for any new transactions going
+forward.
+
+This means that from a philosophical viewpoint, there ought to be no reason that Tari Script is not workable, and
+further, that pruning spent outputs (and possibly the scripts associated with them) is no different from pruning range
+proofs.
+
+### Additional requirements
+
+The assumptions that broadly equate scripting with range proofs in the above argument are:
+
+* The script (hash) must be committed to in the blockchain.
+* The script must not be malleable in any way without invalidating the transaction.
+* The creator of the UTXO must commit to and sign the script (hash).
+
+The script commitment, which can be adequately represented by the hash of the canonical serialisation of the script in
+binary format, could be placed in the transaction kernel, or in a dedicated merkle mountain range for scripts.
+
+Range proofs are not malleable because one must have knowledge of the UTXO blinding factor in order to generate a valid
+range proof. However, it's trivial to replace scripts with other valid scripts, potentially to the point that miners or
+malicious actors could take the UTXO for themselves.
+
+Therefore, it's imperative that the UTXO creator sign the script.
+
+Further, this signature must be present in the _kernel_ in some form, otherwise miners will be able to remove the script
+via cut-through, whereas kernels are never pruned.
+
+One approach to commit to the script hashes is to modify the output commitments using the [data commitments] approach
+suggested by [Phyro](https://github.com/phyro). In this approach, when creating a new UTXO, the owner also calculates
+the hash of the locking script, _s_, such that `s = H(script)`. The hash script gets stored in the UTXO itself.
+
+## Protocol modifications
+
+The current definition of a Tari UTXO is:
+
+```rust,ignore
+pub struct TransactionOutput {
+    /// Options for an output's structure or use
+    features: OutputFeatures,
+    /// The homomorphic commitment representing the output amount
+    commitment: Commitment,
+    /// A proof that the commitment is in the right range
+    proof: RangeProof,
+}
+```
+
+Under Tari script, this would change to
+
+```rust,ignore
+pub struct TransactionOutput {
+    features: OutputFeatures,
+    commitment: Commitment,
+    proof: RangeProof,
+    /// New: The hash of the locking script on this UTXO.
+    script_hash: HashOutput,   
+}
+```
+
+Now when calculating the transaction or block balance, we calculate a different set of commitments. The current commitment,
+
+$$ C = v.H + k.G $$
+
+is modified with a commitment to the script hash, so
+
+$$ \hat{C} = C + \mathrm{H}(C \Vert s).G $$
```

Comment:
This is not explicitly stated, but I take it s = script hash?

Reply:
It is L#164

Reply:
I believe you're right. Nice touch!